test/Atmos/EDMF/closures/mixing_length.jl

```diff
@@ -0,0 +1,107 @@
+#### Mixing length model kernels
+
+function mixing_length(
+    ss::SingleStack{FT, N},
+    m::MixingLengthModel,
+    source::Vars,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+    δ::{FT, N},
+    εt::{FT, N},
+    ) where {FT, N}
+
+    # need to code / use the functions: obukhov_length, ustar, ϕ_m
+    ss.edmf.surface
+    # Alias convention:
+    gm = state
+    en = state
+    up = state.edmf.updraft
+    gm_s = source
+    en_s = source
+    up_s = source.edmf.updraft
+    en_d = state.edmf.environment.diffusive
+
+    z = gm_a.z
+    _grav = FT(grav(param_set))
+    ρinv = 1/gm.ρ
+
+
+    fill!(m.L, 0)
+
+    # precompute
+    en_area  = 1 - sum([up[i].ρa for i in 1:N])*ρinv
+    w_env    = (gm.ρu[3]  - sum([up[i].ρau[3] for i in 1:N]))*ρinv
+    en_e_int = (gm.ρe_int - sum([up[i].ρae_int for i in 1:N]))*ρinv
+    en_q_tot = (gm.ρq_tot - sum([up[i].ρaq_tot for i in 1:N]))*ρinv
+    ∂e_int∂z = en_d.e_int # check if there is a missing ∇
+    ∂q_tot∂z = en_d.q_tot # check if there is a missing ∇
+
+    # TODO: check rank of `en_d.∇u`
+    Shear = en_d.∇u[1].^2 + en_d.∇u[2].^2 + en_d.∇u[3].^2 # consider scalar product of two vectors
+
+    # Thermodynamic local variables for mixing length
+    ts::FT    = PhaseEquil(param_set ,en_e_int, gm.ρ, en_q_tot)
+    Π::FT     = exner(ts)
```

Comment:
```suggestion
    Π     = exner(ts)
```
Type casting is not necessary here. Please just use type-casting to prevent unwanted type-promotion, which should be for numbers and CLIMAParameters.